docs/serialization.rst

```diff
@@ -2,18 +2,27 @@ Serialization
 =============
 
 Data in Redis
--------------
+#############
 pydantic-aioredis uses Redis Hashes to store data. The ```_primary_key_field``` of each Model is used as the key of the hash.
 
 Because Redis only supports string values as the fields of a hash, data types have to be serialized.
 
 Simple data types
------------------
+#################
 Simple python datatypes that can be represented as a string and natively converted by pydantic are converted to strings and stored. Examples
 are ints, floats, strs, bools, and Nonetypes.
 
+Unintentional Type Casting with Union Types
+*******************************************
+
+When using Union types, pydantic will cast the value to the first type in the Union. This can cause unintended type casting. For example, if you have a field
+of type Union[float, int], and you set the value to 1, pydantic will cast the value to a float 1.0. This is an issue with Pydantic. More info can be found in
```

Comment:
Maybe also worth noting here, that in the other direction, for `x: Union[int, float]`, having `x=1.99` in the constructor , not only the type gets cast into int, but the **value** actually changes to 1 (as a necessary consequence).

Reply:
@andrewthetechie : I'm not used to reviewing in github, but I saw that in commit `70b9bd2` , the actual tests are added. But this didn't appear in the review pane above. When I click the "Add your review" in the link below this message, I get the same view as that I already approved.

Reply:
Thanks for the example, I added that as well.

The tests don't show up in the review pane because I added them in another branch without a PR.